### PR TITLE
Improve interactive variables

### PR DESCRIPTION
**Pull request by smimram**
This PR builds on #1493 to add the following.

**The interactive variables can be exposed over http via harbor**. For instance, with
```
x = interactive.float(min=0., max=10., "x", 2.)
y = interactive.float("y", 5.)
s = interactive.string("s", "Hello")
b = interactive.bool("b", true)

interactive.harbor()
```
if you connect to <http://localhost:8080/interactive> you will see the nice little page
interactive
where you can change the interactive values in live.

**The interactive variables can be made persistent.** If you add the line
```
interactive.persistent("vars.json")
```
the value of all the variables are stored in the file and automatically loaded with previous value next time.

I expect that this can be quite useful for tweaking paramerters for audio filters, etc.

### CODE REVIEW

**Comment by martinkirch** [comment]
Interaction over harbor is appealing ! any chance it will ever have an equivalent of telnet commands ?

**Comment by smimram** [comment]
@martinkirch anything is doable at this point :)

What do you have in mind exactly? Do you want to be able to set precisely the value of a specific variable? Or more generally run a telnet command over some kind of console on the webpage?

**Comment by martinkirch** [comment]
@smimram I'd use the general option ... even without a webpage !

Something like `GET /commands` would list available commands in JSON (like telnet's `help`) , and `POST /commands` would allow to submit a command (as in telnet). The rationale is that it's often easier for external programs to write/get json over HTTP than passing commands over telnet. 

But I guess a bonus interface would be welcomed, too :)

File: src/lang/builtins_math.ml [hunk 1]
```diff
@@ -139,6 +139,10 @@ let () =
   add_builtin "min_int" ~cat:Math ~descr:"Minimal representable integer."
     ~flags:[Lang.Hidden] [] Lang.int_t (fun _ -> Lang.int min_int)
 
+let () =
+  add_builtin "infinity" ~cat:Math ~descr:"Float representation of infinity." []
+    Lang.float_t (fun _ -> Lang.float infinity)
```
Comment (toots) [inline src/lang/builtins_math.ml:144]: Why not a constant directly here? There's a `add_builtin_base` for that:
```ocaml
  Lang.add_builtin_base ~category:(string_of_category Liq)
    ~descr:"Liquidsoap version string." "liquidsoap.version"
    Lang.(Ground (Ground.String Configure.version))
    Lang.string_t;
```

**Review by toots — COMMENTED** [review]
Some important comments 🙂

File: libs/interactive.liq [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+
+# @flag hidden
+def interactive.string.set(name, v)
+  interactive.string.ref(name) := v
+  p = !interactive.persistency
+  p()
+end
+
+# @flag hidden
+def add_variable(~name, ~type)
```
[the remark this answers is not in the document]
Comment (smimram) [reply libs/interactive.liq:70]: changed to `interactive.create`

File: libs/interactive.liq [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+    "Syntax error or variable not found."
+  end
+end
+server.register(usage="set <name> = <value>", description="Set the value of a variable.", namespace="var", "set", interactive.set)
+
+# Save the value of all interactive variables in a file.
+# @category Interactive
+# @param fname Name of the file.
+def interactive.save(fname)
+  int = [] # Be ready when we add support for those
```
[the remark this answers is not in the document]
Comment (smimram) [reply libs/interactive.liq:124]: No, integers were not there. At some point I will add those, but I feel too tired right now:)

File: libs/interactive.liq [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+  ignore(int)
+  list.iter(fun (nv) -> try interactive.float.set (fst(nv), snd(nv)) catch _ do log.important(label="interactive.load", "Variable #{fst(nv)} not found.") end, float )
+  list.iter(fun (nv) -> try interactive.bool.set  (fst(nv), snd(nv)) catch _ do log.important(label="interactive.load", "Variable #{fst(nv)} not found.") end, bool  )
+  list.iter(fun (nv) -> try interactive.string.set(fst(nv), snd(nv)) catch _ do log.important(label="interactive.load", "Variable #{fst(nv)} not found.") end, string)
+end
+
+# Make the value of interactive variables persistent: they are loaded from the
+# given file and stored there whenever updated. This function should be called
+# after all interactive variables have been defined (variables not declared yet
+# will not be loaded) .
```
[the remark this answers is not in the document]
Comment (smimram) [reply libs/interactive.liq:147]: After `;` we have ` ` :)

File: libs/interactive.liq [hunk 5]
```diff
@@ -0,0 +1,320 @@
...
+# @category Interactive
+# @param fname Name of the file.
+def interactive.persistent(fname)
+  interactive.load(fname)
+  interactive.persistency := {interactive.save(fname)}
+end
+
+# Expose interactive variables through habor http server.
```
[the remark this answers is not in the document]
Comment (smimram) [reply libs/interactive.liq:155]: Yes, let me open a BR for that: #1499.

**Pull request merged**